pkg/ansible/watches/watches.go

```diff
@@ -103,68 +99,59 @@ func parseLabelSelector(dls tempLabelSelector) metav1.LabelSelector {
-	MatchExpressions []tempRequirement `yaml:"matchExpressions,omitempty"`
+	MatchLabels      map[string]string `json:"matchLabels,omitempty"`
+	MatchExpressions []tempRequirement `json:"matchExpressions,omitempty"`
 }
 
 type tempRequirement struct {
-	Key      string                       `yaml:"key"`
-	Operator metav1.LabelSelectorOperator `yaml:"operator"`
-	Values   []string                     `yaml:"values,omitempty"`
+	Key      string                       `json:"key"`
+	Operator metav1.LabelSelectorOperator `json:"operator"`
+	Values   []string                     `json:"values,omitempty"`
 }
 
-// UnmarshalYAML - implements the yaml.Unmarshaler interface for Watch.
-// This makes it possible to verify, when loading, that the GroupVersionKind
-// specified for a given watch is valid as well as provide sensible defaults
-// for values that were omitted.
-func (w *Watch) UnmarshalYAML(unmarshal func(interface{}) error) error {
-	// Use an alias struct to handle complex types
-
-	type alias struct {
-		Group                       string                    `yaml:"group"`
-		Version                     string                    `yaml:"version"`
-		Kind                        string                    `yaml:"kind"`
-		Playbook                    string                    `yaml:"playbook"`
-		Role                        string                    `yaml:"role"`
-		Vars                        map[string]interface{}    `yaml:"vars"`
-		MaxRunnerArtifacts          int                       `yaml:"maxRunnerArtifacts"`
-		ReconcilePeriod             string                    `yaml:"reconcilePeriod"`
-		ManageStatus                bool                      `yaml:"manageStatus"`
-		WatchDependentResources     bool                      `yaml:"watchDependentResources"`
-		WatchClusterScopedResources bool                      `yaml:"watchClusterScopedResources"`
-		Blacklist                   []schema.GroupVersionKind `yaml:"blacklist"`
-		Finalizer                   *Finalizer                `yaml:"finalizer"`
-		Selector                    tempLabelSelector         `yaml:"selector"`
-	}
-	var tmp alias
+// Use an alias struct to handle complex types
+type alias struct {
+	Group                       string                    `json:"group"`
+	Version                     string                    `json:"version"`
+	Kind                        string                    `json:"kind"`
+	Playbook                    string                    `json:"playbook"`
+	Role                        string                    `json:"role"`
+	Vars                        map[string]interface{}    `json:"vars"`
+	MaxRunnerArtifacts          int                       `json:"maxRunnerArtifacts"`
+	ReconcilePeriod             metav1.Duration           `json:"reconcilePeriod"`
+	ManageStatus                *bool                     `json:"manageStatus,omitempty"`
+	WatchDependentResources     bool                      `json:"watchDependentResources"`
```

Comment:
Se that the WatchDependentResources is true by default as well.  See;
https://github.com/operator-framework/operator-sdk/blob/16e89924addf13eb4fcf12e3084ccec3e20db0cc/pkg/ansible/watches/watches.go#L73

 So, if it not be informed in the watches file its value will be false when the Unmarshal which will result in a Watch with WatchDependentResources == fase when should be true. It is the same situation which motivates you to add the omitempty for the ManageStatus. See that it is passing in the tests which means that it this scenario is not covered.

IMO we need use the omitempty here for al values that can be nil in the watches. @fabianvf wdyt?